### PR TITLE
Updates to DSC_CONFIG_ROOT env var

### LINKED ISSUE #317: `DSCConfigRoot` should be absolute and not relative
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the latest version
- [X] Search the existing issues.

### Steps to reproduce

dsc -l debug config get -p ./dsc/examples/osinfo.dsc.json

### Expected behavior

```console
2024-02-15T23:35:29.425750Z DEBUG dsc: 36: Running dsc 3.0.0-alpha.5
2024-02-15T23:35:29.425784Z DEBUG dsc::util: 419: Setting 'DSCConfigRoot' env var as '/Users/steve/repos/DSC/dsc/examples'
```

### Actual behavior

```console
2024-02-15T23:35:29.425750Z DEBUG dsc: 36: Running dsc 3.0.0-alpha.5
2024-02-15T23:35:29.425784Z DEBUG dsc::util: 419: Setting 'DSCConfigRoot' env var as './dsc/examples'
```

### Error details

_No response_

### Environment data

```PowerShell
Name                           Value
----                           -----
PSVersion                      7.5.0-preview.1
PSEdition                      Core
GitCommitId                    7.5.0-preview.1
OS                             Darwin 23.3.0 Darwin Kernel Version 23.3.0: Wed Dec 20 21:30:27 PST 2023; root:xnu-10002.81.5~7/RELEASE_ARM64_T8103
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Version

3.0.0-alpha.5

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by anmenaga**
# PR Summary

Fix #317 , #335 

## PR Context

Updates:

1. Renamed env var to be `DSC_CONFIG_ROOT`
2. Added a warning if `DSC_CONFIG_ROOT` env var already exists by the time it is overwritten by `dsc.exe`
3. Path in `DSC_CONFIG_ROOT` env var is now always absolutized
4. The `dsc.exe --path` parameter value is now always absolutized

Before change:
```
PS C:\DSCv3> dsc -l debug config get --path .\..\D1/test_config.dsc.yaml
2024-03-04T20:46:30.706571Z DEBUG dsc: 36: Running dsc 3.0.0-alpha.6
2024-03-04T20:46:30.706935Z DEBUG dsc::util: 419: Setting 'DSCConfigRoot' env var as '.\..\D1'
2024-03-04T20:46:30.707272Z DEBUG dsc::util: 379: Reading input from file .\..\D1/test_config.dsc.yaml
```

After change:
```
PS C:\DSCv3> dsc -l debug config get --path .\..\D1/test_config.dsc.yaml
2024-03-04T20:35:11.929049Z DEBUG dsc: 36: Running dsc 3.0.0-alpha.6
2024-03-04T20:35:11.929408Z DEBUG dsc::util: 436: Setting 'DSC_CONFIG_ROOT' env var as 'C:\D1'
2024-03-04T20:35:11.929742Z DEBUG dsc::util: 380: Reading input from file C:\D1\test_config.dsc.yaml
```

### CODE REVIEW

File: dsc/src/subcommand.rs [hunk 1]
```diff
@@ -205,9 +205,15 @@ pub fn config(subcommand: &ConfigSubCommand, parameters: &Option<String>, stdin:
...
-            set_dscconfigroot(&config_path);
-            get_input(document, stdin, path)
+            let mut new_path = path;
+            let opt_new_path;
+            if path.is_some()
+            {
+                let config_path = path.clone().unwrap_or_default();
+                opt_new_path = Some(absolutize_and_set_dscconfigroot(&config_path));
+                new_path = &opt_new_path;
+            }
```
Comment (SteveL-MSFT) [inline dsc/src/subcommand.rs:215]: I think you can do something like:

```rust
let new_path = if path.is_some() {
  let config_path = path.clone().unwrap_or_default();
  Some(absolutize_and_set_dscconfigroot(&config_path))
} else {
  path
}
```

File: dsc/src/util.rs [hunk 2]
```diff
@@ -406,16 +407,30 @@ pub fn get_input(input: &Option<String>, stdin: &Option<String>, path: &Option<S
     parse_input_to_json(&value)
 }
 
-pub fn set_dscconfigroot(config_path: &str)
+pub fn absolutize_and_set_dscconfigroot(config_path: &str) -> String
```
Comment (SteveL-MSFT) [inline dsc/src/util.rs:410]: I think this function should just be called `set_dscconfigroot` and making it absolute is just a comment than part of the name

File: dsc/src/util.rs [hunk 3]
```diff
@@ -406,16 +407,30 @@ pub fn get_input(input: &Option<String>, stdin: &Option<String>, path: &Option<S
...
 {
     let path = Path::new(config_path);
-    let config_root = match path.parent()
-    {
-        Some(dir_path) => { dir_path.to_str().unwrap_or_default().to_string()},
-        _ => String::new()
+
+    // make path absolute
+    let full_path = path.absolutize().unwrap_or_default();
```
Comment (SteveL-MSFT) [inline dsc/src/util.rs:415]: what does `default` result in here?  should we error instead if we can't get an absolute path?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: dsc/src/subcommand.rs (hunk 1 above)
Comment (anmenaga) [reply dsc/src/subcommand.rs:215]: Hitting
```
temporary value dropped while borrowed
consider using a `let` binding to create a longer lived value
```

File: dsc/src/util.rs (hunk 2 above)
Comment (anmenaga) [reply dsc/src/util.rs:410]: Updated.

File: dsc/src/util.rs (hunk 3 above)
Comment (anmenaga) [reply dsc/src/util.rs]: Yes, processing error is better; updated.

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**